### DISCUSSION TITLE
deploy socket io as https

### DISCUSSION THREAD

**Discussion by sree9578** [miguelgrinberg/Flask-SocketIO#1780] [Q&A]
`socketio.run(app,host=os.getenv('IP', '0.0.0.0'), port =int(os.getenv('PORT', 1001)), ssl_context='adhoc')`
@miguelgrinberg 

i need to run this as https ...
i am getting an error that ssl_context parameter is not there for socket io
So how can i enable ssl for socket io.run ?
should i create certificate and pass it..if it works what will be the parameter for socketio.run()
Feel free to rewrite socketio.run()

thanks

**Comment by miguelgrinberg** [comment]
What web server are you using? The `socketio.run()` function does not expose every possible option to start your web server, it is provided as a convenience with the most common options. But you can always run your web server directly and configure it fully.

**Reply by sree9578** [reply]
i am using wsgi server ...is there a way to add ssl in `socketio.run()`  eventlet server...
if not how can i configure the server directly with ssl in this case?

**Reply by miguelgrinberg** [reply]
For eventlet you can pass the SSL arguments to wrap_socket as additional arguments in your `socketio.run()` call. For example:

    socketio.run(app,host=os.getenv('IP', '0.0.0.0'), port =int(os.getenv('PORT', 1001)), certfile='x.crt', keyfile='x.key')

Or you can start your own WSGI server directly via eventlet, (and use the eventlet docs for SSL setup) or through gunicorn (and use the gunicorn docs for SSL setup).

**Reply by sree9578** [reply]
```
api-service_1  | (21) accepted ('172.18.0.1', 55870)
api-service_1  | Traceback (most recent call last):
api-service_1  |   File "/usr/local/lib/python3.9/dist-packages/eventlet/hubs/poll.py", line 111, in wait
api-service_1  |     listener.cb(fileno)
api-service_1  |   File "/usr/local/lib/python3.9/dist-packages/eventlet/greenthread.py", line 221, in main
api-service_1  |     result = function(*args, **kwargs)
api-service_1  |   File "/usr/local/lib/python3.9/dist-packages/eventlet/wsgi.py", line 837, in process_request
api-service_1  |     proto.__init__(conn_state, self)
api-service_1  |   File "/usr/local/lib/python3.9/dist-packages/eventlet/wsgi.py", line 350, in __init__
api-service_1  |     self.handle()
api-service_1  |   File "/usr/local/lib/python3.9/dist-packages/eventlet/wsgi.py", line 383, in handle
api-service_1  |     self.handle_one_request()
api-service_1  |   File "/usr/local/lib/python3.9/dist-packages/eventlet/wsgi.py", line 412, in handle_one_request
api-service_1  |     self.raw_requestline = self._read_request_line()
api-service_1  |   File "/usr/local/lib/python3.9/dist-packages/eventlet/wsgi.py", line 395, in _read_request_line
api-service_1  |     return self.rfile.readline(self.server.url_length_limit)
api-service_1  |   File "/usr/lib/python3.9/socket.py", line 704, in readinto
api-service_1  |     return self._sock.recv_into(b)
api-service_1  |   File "/usr/local/lib/python3.9/dist-packages/eventlet/green/ssl.py", line 253, in recv_into
api-service_1  |     return self._base_recv(nbytes, flags, into=True, buffer_=buffer)
api-service_1  |   File "/usr/local/lib/python3.9/dist-packages/eventlet/green/ssl.py", line 268, in _base_recv
api-service_1  |     read = self.read(nbytes, buffer_)
api-service_1  |   File "/usr/local/lib/python3.9/dist-packages/eventlet/green/ssl.py", line 187, in read
api-service_1  |     return self._call_trampolining(
api-service_1  |   File "/usr/local/lib/python3.9/dist-packages/eventlet/green/ssl.py", line 162, in _call_trampolining
api-service_1  |     return func(*a, **kw)
api-service_1  |   File "/usr/lib/python3.9/ssl.py", line 1099, in read
api-service_1  |     return self._sslobj.read(len, buffer)
api-service_1  | ssl.SSLError: [SSL: HTTP_REQUEST] http request (_ssl.c:2622)
api-service_1  | Removing descriptor: 7
```

in UI i am getting the socket connection refused ..its connecting to wss.localhost(ip)   ...and failing


may i know what causes this error 

**i am getting these errors when i try with self signed certificates created fom openssl**

**Reply by miguelgrinberg** [reply]
Unfortunately I don't have much experience with eventlet's SSL implementation. But I have seen many other developers having issues.

I personally do not like the idea of involving the Python web server with SSL. I always handle SSL through nginx in front of the Python web server.

**Reply by sree9578** [reply]
even when i try hosting http server with eventlet   

`Mixed Content: The page at '[https://localhost.:4200/#/'](https://localhost:4200/#/%27) was loaded over HTTPS, but requested an insecure XMLHttpRequest endpoint This request has been blocked; the content must be served over HTTPS.`

in backend i am receiving `'bad request'`


i am just trying to emit from server and receive it in angular js client ...
i am confused about how to make connection with socket io server from client now...

**Reply by miguelgrinberg** [reply]
This error has nothing to do with eventlet. You have a mix of https and http content on your page. It should all be https.

Note that I did not say you should use http. The https should be handled at the nginx level, before it reaches the Python server.

**Reply by sree9578** [reply]
Thanks @miguelgrinberg  i will look into it..
